features/features_test.go

```diff
@@ -378,59 +378,92 @@ func (s *featureSuite) TestAll(c *C) {
 	defer st.Unlock()
 	tr := config.NewTransaction(st)
 
+	// Use the first feature as a testing feature and overwrite all its values.
+	fakeFeature := features.SnapdFeature(0)
+	is, why := fakeFeature.IsSupported()
+	c.Check(is, Equals, true)
+	c.Check(why, Equals, "")
+	originalFakeCallback := features.FeaturesSupportedCallbacks[fakeFeature]
+	delete(features.FeaturesSupportedCallbacks, fakeFeature)
+	defer func() {
+		if originalFakeCallback == nil {
+			delete(features.FeaturesSupportedCallbacks, fakeFeature)
+		} else {
+			features.FeaturesSupportedCallbacks[fakeFeature] = originalFakeCallback
+		}
+	}()
+
+	// Change default for fake feature (which may not be set) to be disabled if not set
+	originalEnabledWhenUnset := features.FeaturesEnabledWhenUnset[fakeFeature]
+	delete(features.FeaturesEnabledWhenUnset, fakeFeature)
+	defer func() {
+		if originalEnabledWhenUnset {
+			features.FeaturesEnabledWhenUnset[fakeFeature] = originalEnabledWhenUnset
+		}
+	}()
+
 	allFeaturesInfo := features.All(tr)
 
 	// Feature flags are included even if value unset
-	layoutsInfo, exists := allFeaturesInfo[features.Layouts.String()]
+	fakeFeatureInfo, exists := allFeaturesInfo[fakeFeature.String()]
```

Comment:
If you were to mock `KnownFeatures`, then you could check that `features.All` contains exactly the features you added, rather than just checking the attributes of one feature.